kontonummer.js

```diff
@@ -20,378 +20,440 @@
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'IKANO Banken',
+        regex   : /^(917[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'JAK Medlemsbank',
+        regex   : /^(967[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    },{
+        name    : 'Landshypotek',
+        regex   : /^(939[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    },{
+        name    : 'Lån och Spar Bank Sverige',
+        regex   : /^(963[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'Länsförsäkringar Bank',
+        regex   : /^(340[0-9]|906[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'Länsförsäkringar Bank',
+        regex   : /^(902[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    },{
+        name    : 'Marginalen Bank',
+        regex   : /^(923[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'Nordax Bank',
+        regex   : /^(964[0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    },{
+        name    : 'Nordea',
+        regex   : /^(11[0-9][0-9]|1[4-9][0-9][0-9]|20[0-9][0-9]|3[0-2][0-9][0-9]|330[1-9]|33[1-9][0-9]|34[1-9][0-9]|3[5-9][0-9][0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 10
+        }
+    },{
+        name    : 'Nordea',
+        regex   : /^(4[0-9][0-9][0-9])([0-9]+)$/,
+        modulo  : 11,
+        lengths : {
+            clearing : 4,
+            account  : 7,
+            control  : 11
+        }
+    },{
+        name    : 'Nordea',
+        regex   : /^(3300|3782)([0-9]+)$/,
+        modulo  : 10,
+        lengths : {
+            clearing : 4,
+            account  : 10,
+            control  : 10
+        }
+    },{
+        name    : 'Nordea',
+        regex   : /^([0-9][0-9])(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])([0-9]+)$/,
+        modulo  : 10,
+        lengths : {
+            clearing : 0,
+            account  : 10,
+            control  : 10
+        }
```

Comment:
I'm not sure we want this Nordea validation rule here. This one basically accepts a personnummer (national ID) without a clearing number. It is how Nordea internally usually shows the customers account number, but normally for most kind of money transfers from other banks to Nordea you should enter the clearing number `3300` before the personnummer. Some banks do have these kind of exceptions for Nordea, but I don't think it's safe to assume all do. Nordea is fairly clear in their FAQ that the clearing number `3300` should be used for transfers from other banks... https://www.nordea.se/privat/kundservice/fragor-svar-konton-clearingnummer.html#faq=Clearingnummer

I think it should be merged with the `/^(3300|3782)([0-9]+)$/` above, which in turn is vary lax in its validation rules, to be something like:

```js
regex   : /^(3300|3782)([0-9][0-9])(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])([0-9]+)$/,
```

On the other hand it may be an issue to have a stricter regex, since it's used kind of used for detection rather than validation. When I think about it, I think it would make sense to actually have two different regexes - one for detecting the clearing number to match bank and another for actually validating the account number part in addition to the modulo check, which may return `true` even if the for example the national ID is incorrect.

We don't necessarily need to separate regex as part of this, but it could be an improvement for later. However, I think we should fix this Nordea regex rule.

Reply:
If one really wants to do it properly, a personnummer validation should be validated as date for the first 6/8 digits. A regex will have false positives like 31st of February.

Reply:
@dawaa What do we want to do about this Nordea personnummer regex?

Reply:
I think the first solution you proposed is good enough for now, combining the less strict regex:
```js
/^(3300|3782)([0-9]+)$/
```

with the one that could allow for a personnummer to be a valid input:
```js
/^([0-9][0-9])(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])([0-9]+)$/
```

Because what you mentioned in your most recent comment on "how to do it properly", I'm guessing we would need another regex to somehow correctly check the date. As you mentioned 31st of Feb would be a false positive and so would other dates as well. Meaning we would need the `Date()` API to check if the date is valid or not.

What do you say, do you agree on going further with your first proposal?

Reply:
@dawaa Yes, I think as long as you are required to enter the clearing number it's fine, so combining the regexes will work.